### PR TITLE
Upgrade cakephp

### PR DESCRIPTION
**Pull request by josegonzalez**
- upgrade to minimum cakephp 3.4
- use non-deprecated methods for data retrieval
- add a few scrutinizer fixes
- support all the official versions of php that cakephp supports

### CODE REVIEW

File: src/Listener/ViewSearchListener.php [hunk 1]
```diff
@@ -43,11 +43,12 @@ public function implementedEvents()
      */
     public function afterPaginate(Event $event)
     {
+        $event;
```
Comment (ADmad) [inline src/Listener/ViewSearchListener.php]: ?

File: src/View/Helper/CrudViewHelper.php [hunk 2]
```diff
@@ -326,7 +335,7 @@ public function createRelationLink($alias, $relation, $options = [])
                 'action' => 'add',
                 '?' => [
                     $relation['foreignKey'] => $this->getViewVar('primaryKeyValue'),
-                    '_redirect_url' => $this->request->here
+                    '_redirect_url' => $this->request->getUri()->getPath()
```
Comment (ADmad) [inline src/View/Helper/CrudViewHelper.php]: I am not sure if this returns same path as request->here() which isn't deprecated.

File: src/Listener/ViewSearchListener.php (hunk 1 above)
Comment (josegonzalez) [inline src/Listener/ViewSearchListener.php]: Scrutinizer.

File: src/View/Helper/CrudViewHelper.php (hunk 2 above)
Comment (josegonzalez) [inline src/View/Helper/CrudViewHelper.php]: Thats what `$here` says to use.

File: src/Listener/ViewSearchListener.php (hunk 1 above)
Comment (ADmad) [inline src/Listener/ViewSearchListener.php]: Not a fan of adding silly things to code to please machines.

File: src/View/Helper/CrudViewHelper.php (hunk 2 above)
Comment (ADmad) [inline src/View/Helper/CrudViewHelper.php]: Must be same in that case.

**Pull request merged**